openff/qcsubmit/factories.py

```diff
@@ -125,55 +133,49 @@ def add_workflow_component(
             if issubclass(type(component), CustomWorkflowComponent):
                 try:
                     component.is_available()
+                    self.workflow.append(component)
                 except ModuleNotFoundError as e:
                     raise ComponentRequirementError(
                         f"The component {component.component_name} could not be added to "
                         f"the workflow due to missing requirements"
                     ) from e
-                if component.component_name not in self.workflow.keys():
-                    self.workflow[component.component_name] = component
-                else:
-                    # we should increment the name and add it to the workflow
-                    if "@" in component.component_name:
-                        name, number = component.component_name.split("@")
-                    else:
-                        name, number = component.component_name, 0
-                    # set the new name
-                    component.component_name = f"{name}@{int(number) + 1}"
-                    self.workflow[component.component_name] = component
 
             else:
                 raise InvalidWorkflowComponentError(
                     f"Component {component} rejected as it is not a sub "
                     f"class of CustomWorkflowComponent."
                 )
 
-    def get_workflow_component(self, component_name: str) -> CustomWorkflowComponent:
+    def get_workflow_component(self, component_name: str) -> List[C]:
```

Comment:
Hmm maybe this should now be `get_workflow_components`?

Reply:
Yeah that's a good point and makes more sense with the return type now.